### ISSUE TITLE
Examples?

### ISSUE THREAD

**Issue by kewaopcode** [prikhi/paginate#1]
This package is kinda hard to understand without examples. Will you add them?

**Comment by prikhi** [comment]
Hey! I don't really write much Elm these days, but I have an open source site in production that uses this library that you can reference. You can see the results here:
https://southernexposure.com/categories/tomatoes/

Here I use the querying/rendering functions to create a standard site-specific `Pager` module:
https://github.com/Southern-Exposure-Seed-Exchange/southernexposure.com/blob/develop/client/src/Views/Pager.elm

Which can then be used to generate a pager for some `Paginated` data:
https://github.com/Southern-Exposure-Seed-Exchange/southernexposure.com/blob/81ce4170c4228ab7c580e5861c2a2cf8c814c064/client/src/Products/Views.elm#L153
A "pager" is just a bunch of lazy view functions I call elsewhere:
https://github.com/Southern-Exposure-Seed-Exchange/southernexposure.com/blob/81ce4170c4228ab7c580e5861c2a2cf8c814c064/client/src/Products/Views.elm#L218

Here's the `Config` that specifies how to fetch paginated data:
https://github.com/Southern-Exposure-Seed-Exchange/southernexposure.com/blob/81ce4170c4228ab7c580e5861c2a2cf8c814c064/client/src/PageData.elm#L238

Here's where I fetch the initial data for a page:
https://github.com/Southern-Exposure-Seed-Exchange/southernexposure.com/blob/81ce4170c4228ab7c580e5861c2a2cf8c814c064/client/src/Main.elm#L154-L167

And how I handle the Pagination messages:
https://github.com/Southern-Exposure-Seed-Exchange/southernexposure.com/blob/81ce4170c4228ab7c580e5861c2a2cf8c814c064/client/src/Main.elm

Basically, store the `Paginated` data in your Model, define a `Config` to specify how to fetch additional pages, use the `Retrieving Data` & `Rendering` functions to render the items & a pager, use the `Modifying` & `Updating` functions to manipulate the `Paginated` state: https://package.elm-lang.org/packages/prikhi/paginate/latest/Paginate
